src/ui/resource/Image.js

```diff
@@ -386,25 +392,37 @@ exports = Class(lib.PubSub, function () {
 		return _filterCanvas;
 	};
 
-	this._renderMultiply = function (ctx, srcX, srcY, srcW, srcH, color) {
-		// multiply rgb channels
-		var imgData = this.getImageData(srcX, srcY, srcW, srcH);
-		var data = imgData.data;
-		// simplified multiply math outside of the massive for loop
-		var a = color.a;
-		var mr = 1 + a * ((color.r / 255) - 1);
-		var mg = 1 + a * ((color.g / 255) - 1);
-		var mb = 1 + a * ((color.b / 255) - 1);
-		for (var i = 0, len = data.length; i < len; i += 4) {
-			data[i] *= mr;
-			data[i + 1] *= mg;
-			data[i + 2] *= mb;
+	var unusedCanvas = null;
+
+	this._getTintedImage = function(srcX, srcY, srcW, srcH, color) {
+		var colorString = (color.r << 16 | color.g << 8 | color.b).toString(16);
```

Comment:
let's move this to a separate file (we should move all filter code into separate files, might as well start now)